### PR TITLE
PyInstaller: update to 6.9.0.

### PR DESCRIPTION
**Pull request by Eloitor**
#### Testing the changes
- I tested the changes in this PR: **NO**

#### Local build testing
- I built this PR locally for my native architecture, ( - )

### CODE REVIEW

**Comment by Eloitor** [comment]
I tried to update it and when running it I got:
```
  File "/usr/lib/python3.12/site-packages/pkg_resources/__init__.py", line 937, in _resolve_dist
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pyinstaller-hooks-contrib>=2024.7' distribution was not found and is required by pyinstaller
error: Recipe `build-exe` failed on line 16 with exit code 1
```

I'll not update it soon.

**Comment by Vinfall** [comment]
I think the correct url should be `https://pyinstaller.org/en/stable/CHANGES.html` (instead of `readthedocs.io`). I have no idea why it's still 4.1 (released in 2020!), the latest is already 6.9.0. does any package depends on old releases?

**Comment by Vinfall** [comment]
Quoted from [installation docs](https://pyinstaller.org/en/v6.9.0/installation.html#):

> PyInstaller is split into two packages `pyinstaller` and `pyinstaller-hooks-contrib`. These should be kept approximately (roughly within a year of each other) in sync. To that end, if you choose to pin your `pyinstaller` version, please also pin `pyinstaller-hooks-contrib`. If you upgrade `pyinstaller`, you will also need to upgrade `pyinstaller-hooks-contrib`.

and it's required indeed in [setup.cfg](https://github.com/pyinstaller/pyinstaller/blob/v6.9.0/setup.cfg#L68), I'm not sure why it builds successfully w/o it...Basically you just need to package https://github.com/pyinstaller/pyinstaller-hooks-contrib, mark it as a dep, then rebuild pyinstaller should work.

**Comment by Vinfall** [comment]
I think it's better to update PyInstaller since installation docs mentioned these two should be in sync. 2022 vs 2024 is definitely not the case. I can build and run it successfully with the new subpackage.

You may find this patch useful (I have to use `${PYPI_SITE}` as GitHub URL does not work after updating, did not check changes in build system as it still builds):
```diff
Subject: [PATCH] PyInstaller: update to 6.9.0.

---
 srcpkgs/PyInstaller/template | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/srcpkgs/PyInstaller/template b/srcpkgs/PyInstaller/template
index 474d44d0..14ea950e 100644
--- a/srcpkgs/PyInstaller/template
+++ b/srcpkgs/PyInstaller/template
@@ -1,7 +1,7 @@
 # Template file for 'PyInstaller'
 pkgname=PyInstaller
-version=4.1
-revision=5
+version=6.9.0
+revision=1
 build_style=python3-module
 hostmakedepends="python3-setuptools"
 makedepends="zlib-devel"
@@ -11,8 +11,8 @@ maintainer="Alex Childs <misuchiru03+void@gmail.com>"
 license="GPL-2.0-only"
 homepage="http://www.pyinstaller.org"
 changelog="https://pyinstaller.org/en/stable/CHANGES.html"
-distfiles="https://github.com/pyinstaller/pyinstaller/releases/download/v${version}/pyinstaller-${version}.tar.gz"
-checksum=954ae81de9a4bc096ff02433b3e245b9272fe53f27cac319e71fe7540952bd3d
+distfiles="${PYPI_SITE}/p/pyinstaller/pyinstaller-${version}.tar.gz"
+checksum=f4a75c552facc2e2a370f1e422b971b5e5cdb4058ff38cea0235aa21fc0b378f
 
 CFLAGS="-Wno-stringop-truncation"
 
-- 
2.46.0
```

And running checks return this:
```
=> PyInstaller-6.9.0_1: running do_check ...
=> WARNING: No command 'test' defined by setup.py.
```

File: srcpkgs/PyInstaller/template [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+revision=5
 build_style=python3-module
 hostmakedepends="python3-setuptools"
 makedepends="zlib-devel"
-depends="python3 python3-altgraph"
+depends="python3 python3-altgraph python3-pyinstaller-hooks-contrib"
 short_desc="Bundles a Python application and dependencies into one package"
 maintainer="Alex Childs <misuchiru03+void@gmail.com>"
 license="GPL-2.0-only"
 homepage="http://www.pyinstaller.org"
```
Comment (classabbyamp) [inline srcpkgs/PyInstaller/template:12]: https

**Comment by classabbyamp** [comment]
it's better to use the git source tarball, because the pypi ones often exclude tests

**Pull request closed without merging**